conformance_tests/tools/metrics/src/test_metric.cpp

```diff
@@ -1321,6 +1325,201 @@ TEST_F(
+    zeCommandListAppendLaunchKernel(commandList, function, &tg, nullptr, 0,
+                                    nullptr);
+    lzt::close_command_list(commandList);
+
+    for (auto groupInfo : metricGroupInfo) {
+      LOG_INFO << "test metricGroup name " << groupInfo.metricGroupName;
+      lzt::activate_metric_groups(device, 1, &groupInfo.metricGroupHandle);
+
+      ze_event_handle_t eventHandle;
+      lzt::zeEventPool eventPool;
+      eventPool.create_event(eventHandle, ZE_EVENT_SCOPE_FLAG_HOST,
+                              ZE_EVENT_SCOPE_FLAG_HOST);      
+
+      auto startTime = std::chrono::system_clock::now();
+      zet_metric_streamer_handle_t metricStreamerHandle =
+          lzt::metric_streamer_open_for_device(
+              device, groupInfo.metricGroupHandle, eventHandle,
+              notifyEveryNReports, samplingPeriod);
+      ASSERT_NE(nullptr, metricStreamerHandle);
+
+      lzt::execute_command_lists(commandQueue, 1, &commandList, nullptr);
+      lzt::synchronize(commandQueue, std::numeric_limits<uint64_t>::max());
+
+      double minimumTimeBeforeEventIsExpected = notifyEveryNReports * (static_cast<double>(samplingPeriod) / nanoSecToSeconds);
+      // Initializing a 10% error buffer to prevent corner cases 
+      double errorBuffer = 0.10 * minimumTimeBeforeEventIsExpected;
+      LOG_DEBUG << "minimumTimeBeforeEventIsExpected " << minimumTimeBeforeEventIsExpected;
+      LOG_DEBUG << "errorBuffer " << errorBuffer;
+      
+      ze_result_t eventResult;
+      do {
+        auto currentTime = std::chrono::system_clock::now();
+        std::chrono::duration<double> elapsedSeconds = currentTime - startTime;
+        if (elapsedSeconds.count() >= (minimumTimeBeforeEventIsExpected - errorBuffer)) {
+          break;
+        }
+        eventResult = zeEventQueryStatus(eventHandle);
+        EXPECT_EQ(eventResult, ZE_RESULT_NOT_READY);
+        std::this_thread::sleep_for(std::chrono::milliseconds(1000));
+      } while (true);
+
+      // Sleep again for errorBuffer time to ensure corner cases are avoided.
+      uint32_t sleep = std::ceil(errorBuffer);
+      LOG_DEBUG << "additional sleep before expecting event to be ready " << sleep;
+      std::this_thread::sleep_for(std::chrono::seconds(sleep));
+
+      eventResult = zeEventQueryStatus(eventHandle);
+      EXPECT_EQ(eventResult, ZE_RESULT_SUCCESS);
+
+      lzt::metric_streamer_close(metricStreamerHandle);
+      lzt::deactivate_metric_groups(device);
+      eventPool.destroy_event(eventHandle);     
+    }
+    lzt::destroy_function(function);
+    lzt::free_memory(a_buffer);
+    lzt::free_memory(b_buffer);
+    lzt::free_memory(c_buffer);
+    lzt::reset_command_list(commandList);
+    lzt::destroy_command_queue(commandQueue);
+    lzt::destroy_command_list(commandList);
+  }
+}
+
+TEST_F(
+    zetMetricStreamerTest,
+    GivenValidMetricGroupWhenTimerBasedStreamerIsCreatedThenExpectStreamerToGenrateCorrectNumberOfReports) {
```

Comment:
@matcabral 
To add to the discussion about merging tests: 
there are 2 tests here
WhenTimerBasedStreamerIsCreatedThenExpectStreamerTo**NotifyEventAtProperTime**AndSucceed
WhenTimerBasedStreamerIsCreatedThenExpectStreamerTo**GenrateCorrectNumberOfReports**

Some points related to having 1 test validating 1 behaviour approach is that 
1. we could identify failures from test name
    Example: We currently have "WhenTimerBasedStreamerIsCreatedThenExpectStreamerToSucceed" which is Opening, reading, calculating and validating data. This ideally could be just "StreamerOpen" validation.
2. further extension of a test for that behaviour is possible without impacting any related operations
3. validate more conditions related to the behaviour. For example: readData() could have specific tests which would validate fixed-buffer, Query size +read etc.

On the flip-side, the time for CI run has be considered as well.
Currently the time taken for each test looks too high.
The assumption of 20% of Metric groups may be too high for BMG, which has too many metric groups. 
Each test on BMG seem to take from 19 to 36 seconds (https://gtax-jfc.intel.com/#/jobs/7002092#task_tests_40)

May be we could run tests for 1 metric group from each domain for daily CI and 5% of metric groups for weekly CI ?

Reply:
Thanks for the clarification on the differences of the tests, that was exactly what I was asking for. If clearly distinctive behaviors are covered in each test, then is good to have them separated. 

>May be we could run tests for 1 metric group from each domain for daily CI and 5% of metric groups for weekly CI ?

makes sense avoiding duplicated behavior testing. conceptually one per domains seems like a good approach (after all EU stall and OA are the only ones sharing APIs but different domain), but given that we are stepping away from the domain concept we want to aim at using new concurrent metric groups API.

Reply:
>but given that we are stepping away from the domain concept we want to aim at using new concurrent metric groups API.

Right. Added to https://jira.devtools.intel.com/browse/VLCLJ-2348